nntrainer/graph/network_graph.h

```diff
@@ -153,9 +153,9 @@ class NetworkGraph {
    * @retval Layer
    */
   std::shared_ptr<Layer> getLayer(const std::string &layer_name) {
-    for (auto iter = layers.begin(); iter != layers.end(); ++iter) {
-      if ((*iter)->getName() == layer_name) {
-        return *iter;
+    for (auto iter = adj.begin(); iter != adj.end(); ++iter) {
+      if ((*iter).front().layer->getName() == layer_name) {
```

Comment:
I wonder about the semantics of layer_name, is it `istrequal` or case-sensitive?

Reply:
that's a good question. Layer names are meant to a unique case-sensitive string. So, IMO using `istrequal` goes against that invariant.
This must be done all over code though.

Reply:
It's good, we can leave this as an issue, with good first issue labeled.